crates/chain/src/tx_graph.rs

```diff
@@ -446,25 +947,30 @@ impl Additions {
 
     /// Appends the changes in `other` into self such that applying `self` afterward has the same
     /// effect as sequentially applying the original `self` and `other`.
-    pub fn append(&mut self, mut other: Additions) {
+    pub fn append(&mut self, mut other: Additions<A>)
+    where
+        A: Ord,
+    {
         self.tx.append(&mut other.tx);
         self.txout.append(&mut other.txout);
+        self.anchors.append(&mut other.anchors);
+        self.last_seen.append(&mut other.last_seen);
```

Comment:
What if `other` has a smaller `last_seen` than `self.`. This will decrease `last_seen` but `last_seen` should be monotonically increasing. I think fix and add test.